packages/web-config-server/src/apiV1/dataBuilders/generic/analytics/analyticsPerPeriod.js

```diff
@@ -30,6 +30,7 @@ class AnalyticsPerPeriodBuilder extends DataBuilder {
   }
 
   getResultsPerPeriod = results => {
+    if (!results) return [];
```

Comment:
Curious about this change, what was the reason for it? I think at the least we ought be to returning `{}` no? as that seems to be the shape of `resultsPerPeriod`?

Might be easier to do:
```suggestion
  getResultsPerPeriod = (results = [])=> {
```

Reply:
Yes you are right, it should return {}. 

The reason to add this is that, when frontend requests data, it will request twice (similar behaviour in [this card](https://github.com/beyondessential/tupaia-backlog/issues/2524). When config server is managing the first response, result could be `undefined` from aggregator if there is no data, which will cause error when performing `.forEach()` later, with that, error will be sent to frontend, which is not what we want...

I'm happy to make a new card and investigate it.

Reply:
Sorry I couldn't reproduce now, I guess we have data coming in (a few days ago) and it fixed this issue. I will still create a new card and mention it.